### PR TITLE
One MavenRepository to rule them all? Refs #60

### PR DESCRIPTION
**Pull request by dwijnand**
@eed3si9n is this what you had in mind?

### CODE REVIEW

**Comment by eed3si9n** [comment]
Yea. Pretty much.

File: librarymanagement/src/main/scala/sbt/librarymanagement/Resolver.scala [hunk 1]
```diff
@@ -258,8 +258,8 @@ abstract class ResolverFunctions {
       new File(sbt.io.Path.userHome, ".m2/repository")
   }
   // TODO - should this just be the *exact* same as mavenLocal?  probably...
-  def publishMavenLocal: MavenCache = new MavenCache("publish-m2-local", mavenLocalDir)
-  def mavenLocal: IMavenRepository = new MavenCache("Maven2 Local", mavenLocalDir)
+  def publishMavenLocal: MavenRepository = new MavenRepository("publish-m2-local", mavenLocalDir)
```
Comment (eed3si9n) [inline librarymanagement/src/main/scala/sbt/librarymanagement/Resolver.scala]: We should try to use `apply` when possible to unify towards it.

**Review by eed3si9n — CHANGES_REQUESTED** [review]
Minor comment.

**Comment by dwijnand** [comment]
I don't like this modelling with rootFile being an Option..

**Comment by dwijnand** [comment]
And `root: String` and `rootFile: Option[File]` both being properties for the underlying datatype.

**Review by eed3si9n — APPROVED** [review]

**Comment by eed3si9n** [comment]
I think it's an acceptable compromise on keeping the older `MavenRepository`-like behavior and the feature.
The alternative would be to turn `MavenRepository` into an interface, and come up with a sensible name for non-cache values.

**Comment by dwijnand** [comment]
I prefer the alternative then. Aka option 2 I gave in #60

**Comment by eed3si9n** [comment]
ok. AFAIK the only people using `MavenCache` is https://github.com/sbt/sbt-maven-resolver/blob/503843421fdf18c89e1af387a45ba9e9ddf1f1d7/src/main/scala/sbt/MavenResolverConverter.scala, so I'm fine with either solution as long as `MavenRepository`'s `apply` is provided.

**Comment by dwijnand** [comment]
That easy. Much easier than MavenCache's MatchError'ing all over the shop.

**Pull request merged**